Note the issue was caught also by weekly CI
/__w/zephyr/zephyr/lib/libc/common/source/time/gmtime_r.c:35:55: error: expected ';', ',' or ')' before 'tp'
   35 |                                  struct tm *ZRESTRICT tp)
      |                                                       ^~
/__w/zephyr/zephyr/lib/libc/common/source/time/gmtime_r.c:80:45: error: expected ';', ',' or ')' before 'timep'
   80 | struct tm *gmtime_r(const time_t *ZRESTRICT timep,
      |                                             ^~~~~

--- a/lib/libc/common/source/time/gmtime_r.c
+++ b/lib/libc/common/source/time/gmtime_r.c
@@ -10,6 +10,7 @@
  * http://howardhinnant.github.io/date_algorithms.html#civil_from_days
  */
 
+#include <zephyr/toolchain.h>
 #include <time.h>
 
 /* A signed type with the representation of time_t without its

